PizzaDAO/pizza-smartcontract
# Feature/chainlink api

# Add chainlink API integration

adds the initial rare pizzas contract and everything necessary to run the chainlink integration end to end.

To test this you'll need access to the pizza oven repository and to have it configured properly on your local machine.

To test:
1. follow the instructions in `DeployPizzaV1`
2. make sure your .env files are configured properly (contact me if you dont have them)
3. stand up the api server as part of those instructions
4. run `test/integtration/rarepizzas.test.ts`

Expect: the unit test redeems a box you own, the api processes the request, and posts back the ipfs hash of the data.  Calling `tokenURI` on the contract returns a valid result and/or the pizza is visible on opensea.
